src/Error/ErrorsManager.php

```diff
@@ -74,13 +74,14 @@ public function getLintErrors(): array
     }
 
     /**
-     * Returns errors reported for specified path.
-     *
      * @return list<Error>
      */
-    public function forPath(string $path): array
+    public function popAllErrors(): array
```

Comment:
What is the reasoning behind this change? Why do we prefer implicit collection (with result that can vary depending on the usage - placing of `popAllErrors()` call determines if collection will have only errors from 1 file) over explicit collection based on related path?

Reply:
there is no use case so far that you want to collect errors for different paths, between moment we collect errors in worker command, only one file was processed
(process single file, collect errors, collect single file, collect errors, ...)
so this basically mimics `->events = []` that you do in worker command

Reply:
This is leftover from previous approach, but also something that could allow configurable reporting strategy (after each file vs after chunk). With your approach we somehow lose the ability to support it in the future. Also, as you said [you prefer explicitness](https://github.com/Wirone/PHP-CS-Fixer/pull/5#discussion_r1592641131) - IMHO retrieving errors for specific file is more explicit 😅.

Reply:
I've reverted this, if you still feel the need to change it, let's continue in PR#7777.